docs/v3/core/v3.0.rst

```diff
@@ -1211,40 +1210,42 @@ the following procedure:
 
 4. The chunk array ``A`` is equal to ``EC[0]``.
 
+.. _codec-specification:
+
 Specifying codecs
 -----------------
 
-To allow for flexibility to define and implement new codecs, this
-specification does not define any codecs, nor restrict the set of
-codecs that may be used. Each codec must be defined via a separate
-specification. In order to refer to codecs in array metadata
-documents, each codec must have a unique identifier, which is a URI
-that dereferences to a human-readable specification of the codec. A
-codec specification must declare the codec identifier, and describe
-(or cite documents that describe) the encoding and decoding algorithms
-and the format of the encoded data.
-
-A codec may have configuration parameters which modify the behaviour
-of the codec in some way. For example, a compression codec may have a
-compression level parameter, which is an integer that affects the
-resulting compression ratio of the data. Configuration parameters must
-be declared in the codec specification, including a definition of how
-configuration parameters are represented as JSON.
-
-The Zarr core development team maintains a repository of codec
-specifications, which are hosted alongside this specification in the
-`zarr-specs GitHub repository`_, and which are
-published on the `zarr-specs documentation Web site
-<https://zarr-specs.readthedocs.io/>`_. For ease of discovery, it is
-recommended that codec specifications are contributed to the
-zarr-specs GitHub repository. However, codec specifications may be
-maintained by any group or organisation and published in any location
-on the Web. For further details of the process for contributing a
-codec specification to the zarr-specs GitHub repository, see
-`ZEP 0 <https://zarr.dev/zeps/active/ZEP0000.html>`_ which describes
-the process for Zarr specification changes.
-
-Further details of how codecs are configured for an array are given in the `Array metadata`_ section.
+.. _core-codecs:
+
+Core codecs
+-----------
+
+This spec defines a set of codecs ("core codecs") which all Zarr implementations SHOULD implement in 
+order to ensure a minimal level of interoperability between Zarr implementations.
+Each core codec is each defined by specification documents which are hosted in the
+`zarr-specs GitHub repository`_, and are published on the `zarr-specs documentation web site
+<https://zarr-specs.readthedocs.io/>`_. The list of core codecs is part of the Zarr v3 specification. 
+Changes to the list of core codecs MUST be made via the same protocol used for 
+changing the Zarr v3 specification. Changes to the list of core codecs SHOULD be made carefully and
+in close collaboration with extant Zarr v3 implementations. A new core codec SHOULD be added to the 
```

Comment:
```suggestion
changing the Zarr v3 specification. Changes to the list of core codecs SHOULD be made
in close collaboration with members of the Zarr Implementation Council. A new core codec SHOULD be added to the 
```
- Remove "made carefully", which is subjective. I'm not sure what was intended by "carefully", so perhaps this can be replaced by more concrete terms
- Since the zarr implementation concil contains mechanisms for communicating with implementations, I think makes sense to refer to it here.

Reply:
the Zarr implementation council does not represent all zarr implementations. Also, there are active proposals to remove the ZIC entirely. For both these reasons I think we should not mention it specifically here.

Reply:
In that case I think it's important to point to an agreed upon list of community implementations that get to have a say on the list of core codecs, if this sentence is going to be included. I think it's important to be a bit more watertight here to avoid potential disputes in the future.

Reply:
Relying on such a list presupposes that we cannot dynamically tell who has an active zarr implementation. is this a real concern? realistically, if someone comes to this repo saying "I have a zarr implementation, I have problem X Y and Z", then we should listen to them, whether they are on a list or not.

Reply:
this PR is concerned with facilitating codec development for all zarr implementations, not just those written down on a list at some point in time. If the zarr spec developers find it useful to maintain such a list as a tool, then I consider that an implementation detail of spec development and not something that should be referenced specifically in the spec itself.

Reply:
It is a bit funny because "an agreed upon list of community implementations" is what the ZIC is supposed to be and "Zarr implementation council does not represent all zarr implementations" is a shortcoming of the ZIC, imo.

> Also, there are active proposals to remove the ZIC entirely

At this point, there is no decision to remove it. Even if we change the decision making process in https://github.com/zarr-developers/governance/pull/44, we might still keep the ZIC as a review/advisory body. I think promoting a codec to a core codec is a good candidate for asking the ZIC.

Reply:
> Fair - is that one of the reasons the ZIC is being disbanded (sorry, I don't know the history there).

The ZIC is not necessarily being dissolved, but there is an [active proposals to remove the entity](https://github.com/zarr-developers/governance/pull/44). The main concern (from my POV) is that most of the members are not active in zarr spec development, i.e. the ZIC doesn't really do anything.